Core/Types/ModuleInstallDescriptor.cs

```diff
@@ -65,6 +73,15 @@ internal void DeSerialisationFixes(StreamingContext like_i_could_care)
             {
                 throw new BadMetadataKraken(null, "Install stanzas must only include one of file, find, or find_regexp directives");
             }
+
+            // Make sure only filter or include_only fields exist but not both at the same time
+            var filterCount = new[] { filter, filter_regexp }.Count(i => i != null);
+            var includeOnlyCount = new[] { include_only, include_only_regexp }.Count(i => i != null);
+
+            if (filterCount > 0 && includeOnlyCount > 0)
+            {
+                throw new BadMetadataKraken(null, "Install stanzas can only contain filter or include_only directives, not both");
+            }
```

Comment:
Could we allow both? I could see that being useful.

```
"include_only": "Astrogator.*"
"filter": "*.md"
```

Reply:
Which would take preference? When both a filter and an include_only match, there's no  clear decision. We don't have operator ordering in the spec.

Reply:
I was thinking that if both match, that would be treated as an exclude.

Reply:
I think it is better to use separate install clauses to cover such cases. I think that gives us sufficient flexibility, without the confusing interactions of competing filters.